plugins/provisioners/salt/default_downloader.rb

```diff
@@ -0,0 +1,93 @@
+require 'open-uri'
+require 'digest'
+require_relative "./errors"
+
+module VagrantPlugins
+  module Salt
+    class DefaultDownloader
+      BASE_URL = "https://repo.saltproject.io/bootstrap/stable/"
+      PS1 = "bootstrap-salt.ps1"
+      SH  = "bootstrap-salt.sh"
+      SHA256_SUFFIX = "sha256"
+      SIG_SUFFIX = "asc"
+      KEY_URL = "#{BASE_URL}SALT-PROJECT-GPG-PUBKEY-2023.gpg"
+
+      def initialize(communicator)
+        @communicator = communicator
+        @logger  = Log4r::Logger.new("vagrant::salt::default_downloader")
+      end
+
+      def source_url 
+        @communicator == :winrm ? "#{BASE_URL}#{PS1}" : "#{BASE_URL}#{SH}"
+      end
+
+      def get_bootstrap_script
+        @logger.debug "Downloading bootstrap script from #{source_url}"
+        script_file = download_to_tmpfile(source_url)
+
+        if system("gpg --version")
+          @logger.debug "Verifying gpg signature"
+          verify_signature(script_file)
+        else
+          @logger.debug "Skipping gpg signature verification: gpg not found"
```

Comment:
I wasn't sure what to do if GPG wasn't supported here. Do we want to block the download? Or is falling back to just verifying the SHA256 sum good?